File: miscellaneous/graph-dfs/20230111/graph-dft-stack.php
<?php

declare(strict_types=1);

function graphDft(array &$graph): array
{
    $stack = [0];
    $values = [];

    while (count($stack)) {
        $vertex = array_pop($stack);
        $values[] = $vertex;

        foreach ($graph[$vertex] as $connection) {
            if (!in_array(needle: $connection, haystack: $values)) {
                $stack[] = $connection;
            }
        }
    }

    return $values;
}

File: miscellaneous/graph-dfs/20230111/graph_dft_stack.py
from typing import List


class GraphDft:
    def graph_dft(self, graph: List[List[int]]) -> List[int]:
        queue = [0]
        values: List[int] = []

        while (len(queue)):
            vertex = queue.pop()
            values.append(vertex)

            for connection in graph[vertex]:
                if connection not in values:
                    queue.append(connection)

        return values
